### PR TITLE
pinning numpy for older SKL backwards compat tests

### LINKED ISSUE #779: New numpy==2.0.0 breaks runners
From [broken workflow](https://github.com/microsoft/hummingbird/actions/runs/9619950191/job/26537154808):

44 instances like this:
```
___________________ ERROR collecting tests/test_backends.py ____________________
tests/test_backends.py:13: in <module>
    from sklearn.ensemble import GradientBoostingClassifier
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/sklearn/__init__.py:82: in <module>
    from .base import clone
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/sklearn/base.py:17: in <module>
    from .utils import _IS_32BIT
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/sklearn/utils/__init__.py:19: in <module>
    from .murmurhash import murmurhash3_32
sklearn/utils/murmurhash.pyx:1: in init sklearn.utils.murmurhash
    ???
E   ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject

```

It seems other are having similar issues https://github.com/numpy/numpy/issues/26710 , https://github.com/pandas-dev/pandas/issues/59023 https://github.com/scikit-learn/scikit-learn/discussions/29290

### PR DESCRIPTION
**Pull request by ksaur**
Closes #779

### CODE REVIEW

**Comment by ksaur** [comment]
@mshr-h , @interesaaat is traveling/away currently, can you please approve? thanks! :)

**Review by mshr-h — APPROVED** [review]

**Comment by ksaur** [comment]
@mshr-h  i think I had the permissions wrong, can you please try reapproving/re-reviewing and see if it works? (I verified you now should have write access so your approval should hopefully push this through now)...

**Review by mshr-h — APPROVED** [review]

**Pull request merged**
